http/src/request/channel/message/update_message.rs

```diff
@@ -219,29 +233,78 @@ impl<'a> UpdateMessage<'a> {
         Ok(self)
     }
 
-    /// Set the embed of the message.
+    /// Attaches an embed to the message.
+    ///
+    /// Pass `None` if you want to remove all of the embeds.
+    ///
+    /// The first call of this method will clear all present embeds from a
+    /// message and replace it with the set embed. Subsequent calls will add
+    /// more embeds.
     ///
-    /// Pass `None` if you want to remove the message embed.
+    /// To pass multiple embeds at once, use [`embeds`].
     ///
-    /// Note that if there is no content then you will not be
-    /// able to remove the embed of the message.
+    /// # Errors
+    ///
+    /// Returns an [`UpdateMessageErrorType::EmbedTooLarge`] error type if the
+    /// embed is too large.
+    ///
+    /// [`embeds`]: Self::embeds
     pub fn embed(self, embed: impl Into<Option<Embed>>) -> Result<Self, UpdateMessageError> {
         self._embed(embed.into())
     }
 
     fn _embed(mut self, embed: Option<Embed>) -> Result<Self, UpdateMessageError> {
         if let Some(embed_ref) = embed.as_ref() {
-            if let Err(source) = validate::embed(&embed_ref) {
-                return Err(UpdateMessageError {
-                    kind: UpdateMessageErrorType::EmbedTooLarge {
-                        embed: Box::new(embed.expect("embed is known to be some")),
-                    },
-                    source: Some(Box::new(source)),
-                });
+            validate::embed(&embed_ref)
+                .map_err(|source| UpdateMessageError::embed(source, embed_ref.clone(), None))?;
+        }
+
+        if let Some(embed) = embed {
+            if let Some(embeds) = &mut self.fields.embeds {
+                embeds.push(embed);
+            } else {
+                self.fields.embeds.replace(Vec::from([embed]));
             }
+        } else {
+            self.fields.embeds.replace(Vec::new());
         }
 
-        self.fields.embed.replace(embed);
+        Ok(self)
+    }
+
+    /// Set the embeds of the message.
+    ///
+    /// To keep all embeds, do not use this.
+    ///
+    /// To modify one or more embeds in the message, acquire them from the
+    /// previous message, mutate them in place, then pass that list to this
+    /// method.
+    ///
+    /// To remove all embeds, pass an empty list: `iter::empty`
```

Comment:
The way this is conveyed is a bit odd. Maybe we could go with "To remove all embeds pass an empty iterator via a function like [`std::iter::empty`]."?